core/tests/puppeteer-acceptance-tests/utilities/user/logged-in-user.ts

```diff
@@ -342,6 +368,199 @@ export class LoggedInUser extends BaseUser {
       );
     }
   }
+
+  /**
+   * Adds a lesson to the 'Play Later' list.
+   *
+   * This method searches for a lesson with the given title in the search results,
+   * clicks on it to open it, waits for the 'Add to Play Later' button to appear,
+   * and then clicks on it to add the lesson to the 'Play Later' list.
+   *
+   * @param {string} lessonTitle - The title of the lesson to add to the 'Play Later' list.
+   */
+  async addLessonToPlayLater(lessonTitle: string): Promise<void> {
+    try {
+      const isMobileViewport = await this.isViewportAtMobileWidth();
+      const lessonCardTitleSelector = isMobileViewport
+        ? desktopLessonCardTitleSelector
+        : mobileLessonCardTitleSelector;
+      await this.page.waitForSelector(lessonCardTitleSelector);
+      const searchResultsElements = await this.page.$$(lessonCardTitleSelector);
+      const searchResults = await Promise.all(
```

Comment:
Yes, Promise.all is needed here because the page.evaluate function is asynchronous and returns a Promise and When we're mapping over searchResultsElements, we're creating an array of Promises which all needs to be resolved before going further.

Reply:
We have a 'await'  before promise.all() that takes care of the promise returned by promise.all(). And, promise.all() would wait for all the promises inside by its own?

Reply:
I see. Thank you for explaining!